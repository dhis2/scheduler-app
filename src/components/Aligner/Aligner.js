import React from 'react'
import { node, bool } from 'prop-types'
import styles from './Aligner.module.css'

/**
 * Convenience component to force elements horizontally or vertically and center them along that
 * axis. Aligns horizontally by default.
 */

const Aligner = ({ children, vertical }) => {
    const className = vertical ? styles.vertical : styles.horizontal

    return <div className={className}>{children}</div>
}

Aligner.defaultProps = {
    vertical: false,
}

Aligner.propTypes = {
    children: node.isRequired,
    vertical: bool,
}

export default Aligner
