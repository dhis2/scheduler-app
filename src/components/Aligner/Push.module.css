.right {
    margin-left: auto;
}

.left {
    margin-right: auto;
}

.up {
    margin-bottom: auto;
}

.down {
    margin-top: auto;
}
