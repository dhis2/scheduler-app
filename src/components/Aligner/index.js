import Aligner from './Aligner'
import Push from './Push'

Aligner.Push = Push

export { Aligner }
