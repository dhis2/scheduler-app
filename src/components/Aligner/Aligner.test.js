import React from 'react'
import { shallow } from 'enzyme'
import Aligner from './Aligner'

describe('<Aligner>', () => {
    it('renders correctly', () => {
        const wrapper = shallow(<Aligner>Text</Aligner>)

        expect(wrapper).toMatchSnapshot()
    })

    it('responds to a vertical prop', () => {
        const wrapper = shallow(<Aligner>Text</Aligner>)

        expect(wrapper).toMatchSnapshot()
    })
})
