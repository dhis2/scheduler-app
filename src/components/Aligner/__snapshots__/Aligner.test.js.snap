// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Aligner> renders correctly 1`] = `
<div
  className="horizontal"
>
  Text
</div>
`;

exports[`<Aligner> responds to a vertical prop 1`] = `
<div
  className="horizontal"
>
  Text
</div>
`;
